tests/crm/test_crm.py

```diff
@@ -0,0 +1,881 @@
+    out = duthost.command(cmd)
+    pytest_assert(out["stdout"] != "", "Get Next Hop IP failed. Neighbor not found")
+    nh_ip2 = [item.split()[0] for item in out["stdout"].split("\n") if "REACHABLE" in item][0]
+
+    nexthop_add_cmd = nexthop_add_cmd.format(network=network, nh_ip1=nh_ip1, nh_ip2=nh_ip2)
+    nexthop_del_cmd = nexthop_del_cmd.format(network=network, nh_ip1=nh_ip1, nh_ip2=nh_ip2)
+
+    # Add nexthop group members
+    RESTORE_CMDS["test_crm_nexthop_group"].append(nexthop_del_cmd)
+    logger.info("Add nexthop groups")
+    duthost.command(nexthop_add_cmd)
+
+    # Make sure CRM counters updated
+    time.sleep(CRM_UPDATE_TIME)
+
+    # Get new "crm_stats_nexthop_group_[member]" used and available counter value
+    new_nexthop_group_used, new_nexthop_group_available = get_crm_stats(get_nexthop_group_stats, duthost)
+
+    if group_member:
+        template_resource = 2
+    else:
+        template_resource = 1
+    # Verify "crm_stats_nexthop_group_[member]_used" counter was incremented
+    pytest_assert(new_nexthop_group_used - nexthop_group_used == template_resource, \
+        "\"crm_stats_nexthop_group_{}used\" counter was not incremented".format("member_" if group_member else ""))
+
+    # Verify "crm_stats_nexthop_group_[member]_available" counter was decremented
+    pytest_assert(nexthop_group_available - new_nexthop_group_available >= template_resource, \
+        "\"crm_stats_nexthop_group_{}available\" counter was not decremented".format("member_" if group_member else ""))
+
+    # Remove nexthop group members
+    logger.info("Removing nexthop groups")
+    duthost.command(nexthop_del_cmd)
+
+    # Make sure CRM counters updated
+    time.sleep(CRM_UPDATE_TIME)
+
+    # Get new "crm_stats_nexthop_group_[member]" used and available counter value
+    new_nexthop_group_used, new_nexthop_group_available = get_crm_stats(get_nexthop_group_stats, duthost)
+
+    # Verify "crm_stats_nexthop_group_[member]_used" counter was decremented
+    pytest_assert(new_nexthop_group_used - nexthop_group_used == 0, \
+        "\"crm_stats_nexthop_group_{}used\" counter was not decremented".format("member_" if group_member else ""))
+
+    # Verify "crm_stats_nexthop_group_[member]_available" counter was incremented
+    pytest_assert(new_nexthop_group_available - nexthop_group_available == 0, \
+        "\"crm_stats_nexthop_group_{}available\" counter was not incremented".format("member_" if group_member else ""))
+
+    #Preconfiguration needed for used percentage verification
+    used_percent = get_used_percent(new_nexthop_group_used, new_nexthop_group_available)
+    if used_percent < 1:
+        nexthop_group_num = get_entries_num(new_nexthop_group_used, new_nexthop_group_available)
+
+        # Increase default Linux configuration for ARP cache
+        increase_arp_cache(duthost, nexthop_group_num, 4, "test_crm_nexthop_group")
```

Comment:
Is ARP cache configuration restored after the test is finished?

Reply:
Yes, its done inside "increase_arp_cache" function, it registers restore commands which will be executed on teardown - "RESTORE_CMDS[test_name].append("sysctl -w " + res["stdout"].replace(" ", ""))"